README.md

```diff
@@ -1 +1,177 @@
+
+The application runs on port `:3001` by default when no stage is provided.
+
+### Environments
+
+We use `NODE_ENV` environment variable to set multi-stage builds (region, stages) with the help of dotenv through npm scripts to load the relevant `.env.<NODE_ENV>` file from `./config` folder into the `serverless.yml` file as we don't rely on serverless for deployment.
+If no `NODE_ENV` value is provided when running the scripts, it will default its `NODE_ENV` value to 'development' with the `.env.development` config file.
+
+The defaulted values for 'stage' and 'region' are `'local'`. Please refer to the values provided in the `serverless.yml` file.
+
+The following values can be provided when running the scripts with `NODE_ENV`:
+
+```ts
+// ./config/.env.<NODE_ENV> files
+'local'; // used for local development
+'development'; // used development staging should we wish to require external services
+'test'; // used during test scripts where local services, mocks can be used in conjonction
+```
+
+```ts
+/** Running serverless offline as an example for a specific stage - 'local'.
+* Stage 'local' will be injected in the serverless.yml
+**/
+NODE_ENV=local serverless offline
+
+```
+
+Further details about environment setup can be found in the provided documentation and `env.example` file.
+
+All secrets the secrets are will stored in `AWS Secrets Manager`.
+
+### Scripts
+
+The following scripts are available, for further information please refer to the project `package.json` file:
+
+- <b>start</b>: `npm start` - _launch serverless offline service_
+- <b>dev</b>: `npm run dev` - _run in parallel the service and unit tests in_ `--watch` _mode with live reload_.
+- <b>test</b>: `npm t` - _execute the unit test suite_
+- <b>build</b>: `npm run build` - _bundle the project for production_
+
+### Offline
+
+Serverless-offline with webpack is used to run the project locally. Please use `npm run dev` script to do so. Go to `http://localhost:3001/local/version` to confirm that everything has loaded correctly, you should see that the version is the same as the version in the `package.json`
+
+To test downloading a certificate:
+
+1. Backup your aws credentials file and change the default credentials to the serverless-s3-local ones (`S3RVER`, `S3RVER`)
+1. create a folder in the root of the project called `buckets`
+1. Run `aws --endpoint http://localhost:4569 s3api put-object --bucket cvs-cert-develop --key <your certificate name> --body <location of your certificate including filename>` to put a copy of your certificate into the S3 bucket which is in the `buckets` folder you created.
+1. Change the metadata of your item in the bucket so that the mime type is `application/octet-stream`
+1. Add
+
+```
+{
+  s3ForcePathStyle: true,
+  endpoint: new AWS.Endpoint('http://localhost:4569'),
```

Comment:
Do we want to set this port as a default value and allow to a `S3_PORT` value to be passed in at build time? I am thinking of instances where the ports might clash and we (devs) don't control them